runtimes/runtimes/util/loggingUtil.ts

```diff
@@ -0,0 +1,58 @@
+import { Logging } from '../../server-interface'
+import { Connection, MessageType } from 'vscode-languageserver/node'
+
+export type LogLevel = 'error' | 'warn' | 'info' | 'log' | 'debug'
+
+const logLevelMap: Record<LogLevel, number> = {
+    error: MessageType.Error,
+    warn: MessageType.Warning,
+    info: MessageType.Info,
+    log: MessageType.Log,
+    debug: MessageType.Debug,
+}
+
+export const isLogLevelEnabled = (l1: LogLevel, l2: LogLevel): boolean => {
+    return logLevelMap[l1] >= logLevelMap[l2]
+}
+
+export const isValidLogLevel = (level: LogLevel): boolean => {
+    return ['error', 'warn', 'info', 'log', 'debug'].includes(level)
```

Comment:
You are copy-pasting these values throughout the code - it's much better to use `MessageType` enum.